### ISSUE TITLE
Tippy is positioned at the top left corner for `display: none` elements.

### ISSUE THREAD

**Issue by abhijit-kar** [pablo-abc/felte#16]
Hey @pablo-abc,

I'm using a library called [Slim Select](http://slimselectjs.com/) for Select drop down. It's an advanced select drop down component.

It hides the original `<select>` element, which causes the tippy error reporter to show error message at the top left of the screen.

![Screenshot from 2021-04-03 22-20-24](https://user-images.githubusercontent.com/25662120/113485387-d737e000-94ca-11eb-8c00-e4511b149e34.png)

Is it possible to specify the container, perhaps the label, of a select element as tippy anchor point?

What should be the best course of action?

P.S. Also the tippy doesn't show up on directly hovering over the Slim Select, instead I had to hover over the label to see the error.

**Comment by pablo-abc** [comment]
I'm reading through slim-select's docs. From what I see it hides the `select` element, so the Tippy instance should be added to the `div` used for the visible select. What I don't see from its docs, is there a way to get the `slim-select` element or pass properties to that div itself?

**Comment by abhijit-kar** [comment]
Slim Select picks up styles from the original select, but not everything.

I wish it did, so that I cud have passed a data- something something to make tippy cling to it instead!

Is there no way:
1. To pass params to tippy
2. Or Make it work on a sibling or parent or different id
3. Or Turn it off and me a use:tippyReporter so that I can either use an absolutely positioned div to show it, or attach it to label?
4. Or do I need to use a below the select error message.
5. Or do I use a different reporter selectively for this field.
5. Or modify slim select to get access to what Tippy reporter needs?

I wish default selects were better from the get go. 😢😭

**Comment by pablo-abc** [comment]
Thanks a lot for your patience! Due to the holidays I have not been able to spend much time in the computer. For now, I'm publishing a minor patch that will allow you to opt out of the reporting selectively.

In your inputs you can add `<input data-felte-reporter-tippy-ignore>` to be able to opt out, this should allow you to manually report errors for incompatible fields.

As a more general solution, I've been working on a way to pass each tippy instance individual props. If we do this, then you could pass a custom `target` to it (which from the documentation I think it can not be changed after it's been created?). The ideal scenario would be for this reporter to work without much hassle with any custom input you might find (and let it still be managed by Felte).

As I said previously, I'm currently on holidays, but I'll try to have this feature during the next few days!

**Comment by pablo-abc** [comment]
Just to let you know, @felte/reporter-tippy@0.2.1 has the ignore functionality! I hope to get back to this ASAP.

**Comment by abhijit-kar** [comment]
Hey @pablo-abc,

No rush, take your sweet time. Blog or my website isn't going anywhere.

Enjoy your holidays.

**Comment by pablo-abc** [comment]
Thanks a lot! I'm making progress on this issue.

Also, your issues have been helping a lot make this library better so I've added you as a contributor on the readme following the [All Contributors spec](https://allcontributors.org/docs/en/specification). If you're not OK with this I can remove you. 😅

**Comment by pablo-abc** [comment]
Feature added and [documented](https://felte.dev/docs/reporters#using-tippy-js) in @felte/reporter-tippy@0.2.2

Feel free to test it and let me know if there might be anything missing.

**Comment by abhijit-kar** [comment]
Thank you @pablo-abc for adding me as a contributor, it means a lot.

I will be testing out the update tonight if I can and will update the ticket accordingly.

Upon successful operation, I will separate one of the form from my site into a separate Repo, so as to showcase Tippy to my blog post readers. And I hope to add 📝 emoji to my contribution. 😅

P.S. I learnt a new thing, Contributor's spec, thanks to you. 🙃

**Comment by pablo-abc** [comment]
Thank you for giving this a try!

Btw, I accidentally broke it for nested forms so use @felte/reporter-tippy@0.2.3 instead of 0.2.2 😅

Sure! Let me know and we can link to it on this README!

**Comment by abhijit-kar** [comment]
@pablo-abc,

I got the time to test out the changes now.

```
"@felte/reporter-tippy": "^0.2.4",
"@felte/validator-yup": "^0.2.1",
"felte": "^0.6.4",
```

![Screenshot from 2021-04-07 12-29-53](https://user-images.githubusercontent.com/25662120/113824091-fec5cb80-979c-11eb-94bb-54986e6e0a5f.png)

## Result:
- [x] data-felte-reporter-tippy-ignore
- [x] data-felte-reporter-tippy-trigger-for
- [x] data-felte-reporter-tippy-for
- [ ] data-felte-reporter-tippy-position-for

The position for doesn't work as expected, I tried using the parent div as the position for, then added an exclamation icon to be its anchor, but it didn't work.

As a work around, I used ignore & custom tippy (But after tippy position for starts working, all the problems will be resolved I guess):

![Screenshot from 2021-04-07 12-37-01](https://user-images.githubusercontent.com/25662120/113824950-020d8700-979e-11eb-9939-8252468f0a01.png)

**Comment by pablo-abc** [comment]
Whoops. That's what happens for working nights. Should be fixed in @felte/reporter-tippy@0.2.5. 

Aaand starting from the current patch version on all packages (0.2.4 in the case of @felte/reporter-tippy) I've completely removed the dependency with `Lodash`, making the minified size [less than a third of the previous size in the case of Felte's core package](https://bundlephobia.com/result?p=felte@0.6.4) if you're interested in reducing your bundle size.

**Comment by abhijit-kar** [comment]
Thanks for removing Lodash, I didn't check all the dependencies for the project before using it. Check this out ([You Don't Need Lodash Anyways!](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore)) for native solution.

I'm always up for reducing bundle size, that's why I chose Svelte, URQL, etc. (i.e. Libraries that are small in size, yet pack a punch!) Felte falls under the same category, having pieces like Legos which I can opt out of, is ohsome!!

I'll update the tippy reporter and get back to you.

**Comment by pablo-abc** [comment]
Lodash was a decision taken to speed up the developing process to have an MVP as soon as possible 😅 I've been removing it little by little with solutions such as the one you pointed out. Starting from 0.6.4 Felte is completely dependency free!

**Comment by abhijit-kar** [comment]
Hey @pablo-abc,

I tried the `data-felte-reporter-tippy-position-for`, but it didn't work.

**Comment by pablo-abc** [comment]
@abhijit-kar I've been looking how `slim-select` works specifically and came to this solution:

```html
  <label data-felte-reporter-tippy-trigger-for="cars" for="cars">Cars</label>
  <div data-felte-reporter-tippy-for="cars">
    <select id="cars" bind:value={$data.cars}>
      <option data-placeholder="true" value=""></option>
      <option value="volvo">Volvo</option>
      <option value="saab">Saab</option>
      <option value="mercedes">Mercedes</option>
      <option value="audi">Audi</option>
    </select>
  </div>
```

As you can see it's not using the position prop. The reason for it seems to be that Felte is not able to "see" the `select` at all since it has a `display: none`, so it does not count it as a field at all (might need to look if it's possible to obtain this element even if it's not displayed). This is why it was not working with the position attribute, and won't work for now.

The solution was, as you might have noticed, to treat this as a custom control component.

**Comment by pablo-abc** [comment]
For accessibility this could probably be changed to:

```html
  <span data-felte-reporter-tippy-trigger-for="cars" id="cars-label">Cars</span>
  <div data-felte-reporter-tippy-for="cars" aria-labelledby="cars-label">
    <select id="cars" bind:value={$data.cars}>
      <option data-placeholder="true" value=""></option>
      <option value="volvo">Volvo</option>
      <option value="saab">Saab</option>
      <option value="mercedes">Mercedes</option>
      <option value="audi">Audi</option>
    </select>
  </div>
```

Since there's no `<select>` visible, the `<label>` element with a `for` attribute would not mean anything to screen readers.

**Comment by abhijit-kar** [comment]
@pablo-abc,

Thanks for the tip on accessibility. I will update all slim select ones accordingly.

Perhaps if the target slim select was accessible, I can add the position-for data attribute to it, so that the tippy shows up above it.

Hoping for the best. 🙃

**Comment by abhijit-kar** [comment]
Hello @pablo-abc,

Here's the link to the demo: [Felte Form Demo](https://www.abhijit-kar.com/felte-form-demo/) & [Felte Form Demo Repo](https://github.com/abhijit-kar/felte-form-demo).

**Comment by pablo-abc** [comment]
Sorry about that. Form mutations with custom controls were a bit trickier than I expected. Fixed in @felte/reporter-tippy@0.2.9

**Comment by abhijit-kar** [comment]
Hey @pablo-abc,

I am relieved that the whole form has a consistent error reporting now, thanks to your efforts.

I couldn't have done it without Felte.
